packages/account-api/src/api/multichain/account.ts

```diff
@@ -0,0 +1,258 @@
+import type { KeyringAccount } from '@metamask/keyring-api';
+import { isScopeEqualToAny } from '@metamask/keyring-utils';
+
+import type {
+  MultichainAccountWallet,
+  MultichainAccountWalletId,
+} from './wallet';
+import type { AccountGroup } from '../group';
+import type { AccountGroupProvider } from '../provider';
+import { AccountWalletCategory } from '../wallet';
+
+const MULTICHAIN_ACCOUNT_ID_REGEX = new RegExp(
+  `^${AccountWalletCategory.Entropy}:.*/(?<groupIndex>\\d+)$`,
+  'u',
+);
+
+/**
+ * Selector to query a specific account based on some criteria.
+ */
+export type MultichainAccountSelector<Account extends KeyringAccount> = {
```

Comment:
Maybe is should be called `KeyringAccountSelector` since it will be used to "select" `KeyringAccount` objects that are part of an `AccountGroup`.

Reply:
🤔 Maybe I could go with `AccountSelector` then... Given that we mostly use a generic type `Account` everywhere (which MUST BE a `KeyringAccount` yes)

WDYT?